tests/Silverback.Integration.Tests.E2E/Kafka/CallbacksTests.cs

```diff
@@ -151,6 +156,177 @@ public async Task EndpointsConfiguredCallback_MessagePublished()
+                    services => services
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .AddSingletonBrokerCallbackHandler<KafkaEndOfTopicPartitionReachedCallback>()
+                        .WithConnectionToMessageBroker(
+                            options => options
+                                .AddMockedKafka(options => options.WithDefaultPartitionsCount(2)))
+                        .AddKafkaEndpoints(
+                            endpoints => endpoints
+                                .Configure(
+                                    config =>
+                                    {
+                                        config.BootstrapServers = "PLAINTEXT://tests";
+                                    })
+                                .AddOutbound<TestEventOne>(
+                                    endpoint => endpoint.ProduceTo(DefaultTopicName, 0))
+                                .AddOutbound<TestEventTwo>(
+                                    endpoint => endpoint.ProduceTo(DefaultTopicName, 1))
+                                .AddInbound(
+                                    endpoint => endpoint
+                                        .ConsumeFrom(DefaultTopicName)
+                                        .Configure(
+                                            config =>
+                                            {
+                                                config.GroupId = "consumer1";
+                                                config.EnablePartitionEof = true;
+                                            })))
+                        .AddIntegrationSpyAndSubscriber())
+                .Run();
+
+            var kafkaTestingHelper = Host.ServiceProvider.GetRequiredService<IKafkaTestingHelper>();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+            var callbackHandlers = Host.ScopedServiceProvider
+                .GetServices<IBrokerCallback>()
+                .Where(service => service is KafkaEndOfTopicPartitionReachedCallback)
+                .ToList();
+
+            callbackHandlers.Should().HaveCount(1);
+            callbackHandlers.Cast<KafkaEndOfTopicPartitionReachedCallback>().Should().OnlyContain(handler => handler.EndOfPartitionReachedCount == 2);
+
+            await kafkaTestingHelper.Broker.Consumers[0].DisconnectAsync();
+
+            await publisher.PublishAsync(messageEventOne);
+            await publisher.PublishAsync(messageEventTwo);
+
+            await kafkaTestingHelper.Broker.Consumers[0].ConnectAsync();
+
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            kafkaTestingHelper.Spy.InboundEnvelopes.Should().HaveCount(2);
+            callbackHandlers.Cast<KafkaEndOfTopicPartitionReachedCallback>().Should().OnlyContain(handler => handler.EndOfPartitionReachedCount == 4);
```

Comment:
Please declare a variable instead of repeating `callbackHandlers.Cast<KafkaEndOfTopicPartitionReachedCallback>()` over and over. Also, take the first item instead of working with the collection, since you only have 1 callback handler.

Something like `var endOfPartitionCallbackHandler = (KafkaEndOfTopicPartitionReachedCallback) callbackHandlers.First()`.